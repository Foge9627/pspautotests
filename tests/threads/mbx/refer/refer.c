#include "../sub_shared.h"

SETUP_SCHED_TEST;

#define REFER_TEST(title, mbx, mbxinfo) { \
	int result = sceKernelReferMbxStatus(mbx, mbxinfo); \
	if (result == 0) { \
		printf("%s: OK\n", title); \
	} else { \
		printf("%s: Failed (%X)\n", title, result); \
	} \
}

int main(int argc, char **argv) {
	SceUID mbx = sceKernelCreateMbx("refer1", 0, NULL);
	SceKernelMbxInfo mbxinfo;

	// Crashes.
	//REFER_TEST("NULL info", mbx, NULL);
	REFER_TEST("Normal", mbx, &mbxinfo);

	int sizes[] = {0, 1, 2, 4, 8, 44, 48, 52, 56, 64, 128, 256, 1024, 0xFF00, 0xFF0000, 0x80000001, 0xCCCCCCCC, 0xFFFFFFFF};
	int numSizes = sizeof(sizes) / sizeof(sizes[0]);

	int i, result;
	for (i = 0; i < numSizes; ++i) {
		mbxinfo.size = sizes[i];
		result = sceKernelReferMbxStatus(mbx, &mbxinfo);
		printf("Size %08X => %08X (result=%08X)\n", sizes[i], mbxinfo.size, result);
	}

	sceKernelDeleteMbx(mbx);

	REFER_TEST("NULL", 0, &mbxinfo);
	REFER_TEST("Invalid", 0xDEADBEEF, &mbxinfo);
	REFER_TEST("Deleted", mbx, &mbxinfo);

	BASIC_SCHED_TEST("NULL",
		result = sceKernelReferMbxStatus(0, &mbxinfo);
	);
	BASIC_SCHED_TEST("Refer other",
		result = sceKernelReferMbxStatus(mbx2, &mbxinfo);
	);
	BASIC_SCHED_TEST("Refer same",
		result = sceKernelReferMbxStatus(mbx1, &mbxinfo);
	);

	return 0;
}