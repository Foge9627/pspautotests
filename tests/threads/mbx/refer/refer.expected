Normal: OK
Size 00000000 => 00000000 (result=00000000)
Size 00000001 => 00000034 (result=00000000)
Size 00000002 => 00000034 (result=00000000)
Size 00000004 => 00000034 (result=00000000)
Size 00000008 => 00000034 (result=00000000)
Size 0000002C => 00000034 (result=00000000)
Size 00000030 => 00000034 (result=00000000)
Size 00000034 => 00000034 (result=00000000)
Size 00000038 => 00000034 (result=00000000)
Size 00000040 => 00000034 (result=00000000)
Size 00000080 => 00000034 (result=00000000)
Size 00000100 => 00000034 (result=00000000)
Size 00000400 => 00000034 (result=00000000)
Size 0000FF00 => 00000034 (result=00000000)
Size 00FF0000 => 00000034 (result=00000000)
Size 80000001 => 00000034 (result=00000000)
Size CCCCCCCC => 00000034 (result=00000000)
Size FFFFFFFF => 00000034 (result=00000000)
NULL: Failed (8002019B)
Invalid: Failed (8002019B)
Deleted: Failed (8002019B)
NULL: A1B1C2E1D1F2 (thread=800201B5, main=8002019B)
Refer other: A1B1C2E1D1F2 (thread=800201B5, main=00000000)
Refer same: A1B1C2E1D1F2 (thread=800201B5, main=00000000)
